### PR TITLE
Home_task_on Module3(Functions)

### PR DESCRIPTION
**Pull request by vshokhava**

### CODE REVIEW

File: tasks/functions/functions.js [hunk 1]
```diff
@@ -13,34 +14,61 @@ function add(a, b) {
...
+/*var person = {
+    firstName: "John",
+    lastName : "Dou"};
+    */
+   function getFullName(firstName, lastName) {
+    var obj = {
+      firstName: firstName,
+      lastName: lastName
+  };
+    return(obj.firstName + " " + obj.lastName);
```
Comment (dshaplyko) [inline tasks/functions/functions.js]: you can omit the brackets and write smth like:
return obj.firstName + " " + obj.lastName;

File: tasks/functions/functions.js [hunk 2]
```diff
@@ -13,34 +14,61 @@ function add(a, b) {
...
  * write function that return shortest of words in given array
  * e.g ["one", "two", "three"] should return one
  */
-function getShortest(wordArray) {
+var wordArray = ["one", "two", "three"];
 
-}
+function getShortest(arr){
+  return (
+  arr.reduce((a, b) => a.length <= b.length ? a : b)
```
Comment (dshaplyko) [inline tasks/functions/functions.js]: it's a good thing to use reduce, +1 point from me :)

But let's also try using for loop in your fee time :)

File: tasks/functions/functions.js [hunk 3]
```diff
@@ -13,34 +14,61 @@ function add(a, b) {
...
+};
+getShortest(wordArray);
 
-/**
- * write function that returns word google with given numbers of "o" symbols
+
+ /* write function that returns word google with given numbers of "o" symbols
  * e.g getGoogle(5) should return "gooooogle"
  */
+let letter = 'o'
```
Comment (dshaplyko) [inline tasks/functions/functions.js]: It's better to define the variable INSIDE the function.

File: tasks/functions/functions.js [hunk 4]
```diff
@@ -51,29 +79,48 @@ function getGoogle(n) {
...
+  var person = new getUser("John", "Doe");
+  console.log (person);
+*/
+function getUser(firstName = null, lastName = null, age = null) {
+  var obj = {
+    firstName: firstName,
+    lastName: lastName,
+    age: age
+};
+  return obj;
```
Comment (dshaplyko) [inline tasks/functions/functions.js]: you can simply return the object:
...
return {
 firstName,
 lastName,
 age
}

File: tasks/functions/functions.js [hunk 5]
```diff
@@ -51,29 +79,48 @@ function getGoogle(n) {
...
-function getTotalPath(path) {
-
+const path = [
+    {  direction: "Kiyv - Minsk", distance: 567},
+    {  direction: "Kiyv - Paris", distance: 2402}
+];
+function getTotalPath(arr){
+      let total = 0;
+      let i;
+       for (i = 0; i < arr.length; i++) {
```
Comment (dshaplyko) [inline tasks/functions/functions.js]: it's better to define the iterable variable inside the for loop:
for (let i = 0; i < arr.length; i++)...

File: tasks/functions/functions.js (hunk 1 above)
Comment (vshokhava) [reply tasks/functions/functions.js]: done

File: tasks/functions/functions.js (hunk 2 above)
Comment (vshokhava) [reply tasks/functions/functions.js]: remained as is

File: tasks/functions/functions.js (hunk 3 above)
Comment (vshokhava) [reply tasks/functions/functions.js]: done

File: tasks/functions/functions.js (hunk 4 above)
Comment (vshokhava) [reply tasks/functions/functions.js]: done

File: tasks/functions/functions.js (hunk 5 above)
Comment (vshokhava) [reply tasks/functions/functions.js]: done

**Pull request merged**